src/lua/lua_task.c

```diff
@@ -2811,6 +2828,43 @@ lua_task_get_symbol (lua_State *L)
 }
 
 static gint
+lua_task_get_symbols_all (lua_State *L)
+{
+	struct rspamd_task *task = lua_check_task (L, 1);
+	struct rspamd_metric *metric;
+	struct rspamd_metric_result *mres;
+	GHashTableIter it;
+	gpointer k;
+	gboolean found = FALSE;
+	gint i = 1;
+
+	if (task) {
+		metric = task->cfg->default_metric;
+		mres = g_hash_table_lookup (task->results, DEFAULT_METRIC);
+		if (mres) {
+			found = TRUE;
+			lua_newtable (L);
+			g_hash_table_iter_init (&it, mres->symbols);
+			while (g_hash_table_iter_next (&it, &k, NULL)) {
+				lua_push_symbol_result (L, task, metric, k, FALSE, TRUE);
```

Comment:
I'd recommend to split `lua_push_symbol_result` into two parts: one with hash table lookup and one that accepts the complete struct rspamd_symbol. This will optimise execution slightly.